sdk/data_kusto/src/operations/query.rs

```diff
@@ -0,0 +1,174 @@
+use crate::client::KustoClient;
+use crate::request_options::*;
+use azure_core::prelude::*;
+use azure_core::setters;
+use azure_core::{
+    collect_pinned_stream,
+    headers::{add_mandatory_header2, add_optional_header2},
+    Response as HttpResponse,
+};
+use futures::future::BoxFuture;
+
+/// A future of a delete file response
+type ExecuteQuery = BoxFuture<'static, crate::error::Result<KustoResponseDataSetV2>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct QueryBody {
+    db: String,
+    csl: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecuteQueryBuilder {
+    client: KustoClient,
+    database: String,
+    query: String,
+    client_request_id: Option<ClientRequestId>,
+    context: Context,
+}
+
+impl ExecuteQueryBuilder {
+    pub(crate) fn new(
+        client: KustoClient,
+        database: String,
+        query: String,
+        context: Context,
+    ) -> Self {
+        Self {
+            client,
+            database,
+            query,
+            client_request_id: None,
+            context,
+        }
+    }
+
+    setters! {
+        client_request_id: ClientRequestId => Some(client_request_id),
+        query: String => query,
+        database: String => database,
+        context: Context => context,
+    }
+
+    pub fn into_future(self) -> ExecuteQuery {
+        let this = self.clone();
+        let ctx = self.context.clone();
+
+        Box::pin(async move {
+            let url = this.client.query_url();
+            let mut request = this.client.prepare_request(url, http::Method::POST);
+
+            add_mandatory_header2(
+                &ContentType::new("application/json; charset=utf-8"),
+                &mut request,
+            )
+            .unwrap();
+            add_mandatory_header2(&Accept::new("application/json"), &mut request).unwrap();
```

Comment:
Any reason for all the unwraps in this file?
Using `?` seems to compile without error, and of course won't trigger a panic.

Reply:
this is an artifact from migration from where i developed the code ... the plan is to remove all unwraps throughout the code before merging!

Reply:
moved to new header handling and removed the panics.